plugins/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureRepositoryPlugin.java

```diff
@@ -67,4 +63,16 @@ public AzureRepositoryPlugin(Settings settings) {
             AzureStorageSettings.PROXY_PORT_SETTING
         );
     }
+
+    @Override
+    public boolean reinit(Settings settings) {
+        // secure settings should be readable
+        final Map<String, AzureStorageSettings> clientsSettings = AzureStorageSettings.load(settings);
+        if (clientsSettings.isEmpty()) {
+            throw new IllegalArgumentException("If you want to use an azure repository, you need to define a client configuration.");
+        }
+        assert clientsSettings.containsKey("default") : "always have 'default'";
```

Comment:
I have moved the check as suggested in the below comment, but kept the assert there since it would truly be a code problem instead of an external one.